From 78a4a530ab1d0857f1aa1f9034b4f40fbffcafdc Mon Sep 17 00:00:00 2001
From: Peter Delevoryas <pdel@fb.com>
Date: Wed, 15 Jun 2022 13:45:39 -0700
Subject: [PATCH 518/522] aspeed: fby35: Add bmc, bb, and nic FRU's

This provides support to run `fruid-util bmc`, `fruid-util bb`, and
`fruid-util nic` and get some real (but redacted) output.

    root@bmc-oob:~# fruid-util bmc

    FRU Information           : BMC
    ---------------           : ------------------
    Board Mfg Date            : Mon Jan 10 21:42:00 2022
    Board Mfg                 : XXXXXX
    Board Product             : BMC Storage Module
    Board Serial              : XXXXXXXXXXXXX
    Board Part Number         : XXXXXXXXXXXXXX
    Board FRU ID              : 1.0
    Board Custom Data 1       : XXXXXXXXX
    Board Custom Data 2       : XXXXXXXXXXXXXXXXXX
    Product Manufacturer      : XXXXXX
    Product Name              : Yosemite V3.5 EVT2
    Product Part Number       : XXXXXXXXXXXXXX
    Product Version           : EVT2
    Product Serial            : XXXXXXXXXXXXX
    Product Asset Tag         : XXXXXXX
    Product FRU ID            : 1.0
    Product Custom Data 1     : XXXXXXXXX
    Product Custom Data 2     : Config A
    root@bmc-oob:~# fruid-util bb

    FRU Information           : Baseboard
    ---------------           : ------------------
    Chassis Type              : Rack Mount Chassis
    Chassis Part Number       : N/A
    Chassis Serial Number     : N/A
    Board Mfg Date            : Fri Jan  7 10:30:00 2022
    Board Mfg                 : XXXXXX
    Board Product             : Management Board wBMC
    Board Serial              : XXXXXXXXXXXXX
    Board Part Number         : XXXXXXXXXXXXXX
    Board FRU ID              : 1.0
    Board Custom Data 1       : XXXXXXXXX
    Board Custom Data 2       : XXXXXXXXXXXXXXXXXX
    Product Manufacturer      : XXXXXX
    Product Name              : Yosemite V3.5 EVT2
    Product Part Number       : XXXXXXXXXXXXXX
    Product Version           : EVT2
    Product Serial            : XXXXXXXXXXXXX
    Product Asset Tag         : XXXXXXX
    Product FRU ID            : 1.0
    Product Custom Data 1     : XXXXXXXXX
    Product Custom Data 2     : N/A
    root@bmc-oob:~# fruid-util nic

    FRU Information           : NIC
    ---------------           : ------------------
    Board Mfg Date            : Tue Nov  2 08:51:00 2021
    Board Mfg                 : XXXXXXXX
    Board Product             : Mellanox ConnectX-6 DX OCP3.0
    Board Serial              : XXXXXXXXXXXXXXXXXXXXXXXX
    Board Part Number         : XXXXXXXXXXXXXXXXXXXXX
    Board FRU ID              : FRU Ver 0.02
    Product Manufacturer      : XXXXXXXX
    Product Name              : Mellanox ConnectX-6 DX OCP3.0
    Product Part Number       : XXXXXXXXXXXXXXXXXXXXX
    Product Version           : A9
    Product Serial            : XXXXXXXXXXXXXXXXXXXXXXXX
    Product Custom Data 3     : ConnectX-6 DX

Signed-off-by: Peter Delevoryas <pdel@fb.com>
---
 hw/arm/aspeed.c       |  1 +
 scripts/gen_eeprom.py | 17 +++++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 scripts/gen_eeprom.py

diff --git a/hw/arm/aspeed.c b/hw/arm/aspeed.c
index a3db3406c5..9c54269d68 100644
--- a/hw/arm/aspeed.c
+++ b/hw/arm/aspeed.c
@@ -23,6 +23,7 @@
 #include "qemu/units.h"
 #include "hw/core/qdev-clock.h"
 #include "system/system.h"
+#include "hw/arm/fby35.h"
 
 static struct arm_boot_info aspeed_board_binfo = {
     .board_id = -1, /* device-tree-only board */
diff --git a/scripts/gen_eeprom.py b/scripts/gen_eeprom.py
new file mode 100644
index 0000000000..b5eceb47bd
--- /dev/null
+++ b/scripts/gen_eeprom.py
@@ -0,0 +1,17 @@
+import os
+import sys
+
+prefix = os.environ.get('PREFIX', '')
+
+for bin_file in sys.argv[1:]:
+    var_name = prefix + bin_file.split('.')[0]
+    sys.stdout.write('static const uint8_t %s[] = {' % var_name)
+    with open(bin_file, 'rb') as f:
+        s = f.read()
+        for i, b in enumerate(s):
+            if i % 16 == 0:
+                sys.stdout.write('\n    ')
+            elif i != 0:
+                sys.stdout.write(' ')
+            sys.stdout.write('0x%02x,' % b)
+    sys.stdout.write('\n};\n')
-- 
2.53.0-Meta

